Woops ;)

```
/var/lib/jenkins/jobs/dron/workspace/Nodes/master/ground/gcs/src/plugins/config/configstabilizationwidget.cpp:67:27: error: cannot convert ‘UAVObject*’ to ‘UAVObjectField*’ in assignment
     manualControlSettings = getObjectManager()->getObject(ManualControlSettings::NAME);
```

--- a/ground/gcs/src/plugins/config/configinputwidget.cpp
+++ b/ground/gcs/src/plugins/config/configinputwidget.cpp
@@ -51,6 +51,7 @@
 #include <coreplugin/generalsettings.h>
 
 #include "actuatorcommand.h"
+#include "actuatorsettings.h"
 
 // The fraction of range to place the neutral position in
 #define THROTTLE_NEUTRAL_FRACTION 0.02
@@ -335,6 +336,11 @@ ConfigInputWidget::ConfigInputWidget(QWidget *parent) : ConfigTaskWidget(parent)
                 connect(child, SIGNAL(currentTextChanged(QString)), this, SLOT(checkFlightMode(QString)));
         }
     }
+
+    // display warning if hangtime is enabled but not using switch arming
+    ActuatorSettings *actuatorSettings = qobject_cast<ActuatorSettings *>(getObjectManager()->getObject(ActuatorSettings::NAME));
+    if (actuatorSettings)
+        connect(actuatorSettings, SIGNAL(objectUpdated(UAVObject *)), this, SLOT(checkHangtimeConfig()));
 }
 void ConfigInputWidget::resetTxControls()
 {
@@ -1751,6 +1757,9 @@ void ConfigInputWidget::checkArmingConfig(QString option)
         m_config->lblThrottleCheckWarn->hide();
     else
         m_config->lblThrottleCheckWarn->show();
+
+    // display warning if hangtime is enabled but not using switch arming
+    checkHangtimeConfig();
 }
 
 void ConfigInputWidget::checkFlightMode(QString option)
@@ -1773,4 +1782,18 @@ void ConfigInputWidget::checkFlightMode(QString option)
                 m_config->lblMultiwii->show();
         }
     }
+}
+
+void ConfigInputWidget::checkHangtimeConfig()
+{
+    bool warn = true;
+
+    ActuatorSettings *actuatorSettings = qobject_cast<ActuatorSettings *>(getObjectManager()->getObject(ActuatorSettings::NAME));
+    if (actuatorSettings)
+        warn &= actuatorSettings->getLowPowerStabilizationMaxTime() > 0.0f;
+
+    const QString option = m_config->armControl->currentText();
+    warn &= !option.startsWith("Switch") && option != "Always Disarmed";
+
+    m_config->lblHangTimeWarning->setVisible(warn);
 }

--- a/ground/gcs/src/plugins/config/configinputwidget.h
+++ b/ground/gcs/src/plugins/config/configinputwidget.h
@@ -197,6 +197,7 @@ private slots:
         void updateCalibration();
         void checkArmingConfig(QString option);
         void checkFlightMode(QString option);
+        void checkHangtimeConfig();
 
 protected:
         void resizeEvent(QResizeEvent *event);

--- a/ground/gcs/src/plugins/config/configstabilizationwidget.cpp
+++ b/ground/gcs/src/plugins/config/configstabilizationwidget.cpp
@@ -33,6 +33,7 @@
 
 #include "configstabilizationwidget.h"
 #include "convertmwrate.h"
+#include "manualcontrolsettings.h"
 
 #include <QDebug>
 #include <QStringList>
@@ -49,7 +50,7 @@
 #include <coreplugin/generalsettings.h>
 
 
-ConfigStabilizationWidget::ConfigStabilizationWidget(QWidget *parent) : ConfigTaskWidget(parent)
+ConfigStabilizationWidget::ConfigStabilizationWidget(QWidget *parent) : ConfigTaskWidget(parent), manualControlSettings(nullptr)
 {
     m_stabilization = new Ui_StabilizationWidget();
     m_stabilization->setupUi(this);
@@ -60,6 +61,12 @@ ConfigStabilizationWidget::ConfigStabilizationWidget(QWidget *parent) : ConfigTa
 
     if (!settings->useExpertMode())
         m_stabilization->saveStabilizationToRAM_6->setVisible(false);
+
+    // display switch arming not selected warning when hangtime enabled
+    connect(m_stabilization->sbHangtime, SIGNAL(valueChanged(double)), this, SLOT(hangtimeChanged()));
+    manualControlSettings = getObjectManager()->getObject(ManualControlSettings::NAME);
+    if (manualControlSettings)
+        connect(manualControlSettings, SIGNAL(objectUpdated(UAVObject*)), this, SLOT(hangtimeChanged()));
 
 
     autoLoadWidgets();
@@ -388,4 +395,17 @@ void ConfigStabilizationWidget::showExpoPlot()
       m_stabilization->attitudeStickExpoPlot->plotDataYaw(m_stabilization->attitudeYawExpo->value(), m_stabilization->rateYawKp_3->value(), ExpoCurve::Y_Left);
       update_exp.RateYaw = false;
     }
+}
+
+void ConfigStabilizationWidget::hangtimeChanged()
+{
+    bool warn = m_stabilization->sbHangtime->value() > 0.0;
+    if (manualControlSettings) {
+        UAVObjectField *field = manualControlSettings->getField("Arming");
+        if (field) {
+            const QString option = field->getValue().toString();
+            warn &= !option.startsWith("Switch") && option != "Always Disarmed";
+        }
+    }
+    m_stabilization->lblSwitchArmingWarning->setVisible(warn);
 }

--- a/ground/gcs/src/plugins/config/configstabilizationwidget.h
+++ b/ground/gcs/src/plugins/config/configstabilizationwidget.h
@@ -64,6 +64,8 @@ private:
       bool HorizonRateYaw;
     } update_exp;
 
+    UAVObject *manualControlSettings;
+
 private slots:
     void linkCheckBoxes(int value);
     void processLinkedWidgets(QWidget*);
@@ -73,6 +75,7 @@ private slots:
     void applyMWRateConvertDialog();
 
     void showExpoPlot();
+    void hangtimeChanged();
 };
 
 #endif // ConfigStabilizationWidget_H

--- a/ground/gcs/src/plugins/config/input.ui
+++ b/ground/gcs/src/plugins/config/input.ui
@@ -1405,7 +1405,7 @@ channel value for each flight mode.</string>
             <x>0</x>
             <y>0</y>
             <width>838</width>
-            <height>794</height>
+            <height>817</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
@@ -1486,6 +1486,13 @@ channel value for each flight mode.</string>
                 </property>
                 <property name="wordWrap">
                  <bool>true</bool>
+                </property>
+               </widget>
+              </item>
+              <item>
+               <widget class="QLabel" name="lblHangTimeWarning">
+                <property name="text">
+                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;Warning:&lt;/span&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt; It is strongly recommended to use switch arming when HangTime is enabled.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                </widget>
               </item>

--- a/ground/gcs/src/plugins/config/stabilization.ui
+++ b/ground/gcs/src/plugins/config/stabilization.ui
@@ -615,8 +615,8 @@
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>1006</width>
-            <height>883</height>
+            <width>1008</width>
+            <height>896</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_12">
@@ -10498,8 +10498,8 @@ border-radius: 5;</string>
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>980</width>
-            <height>1019</height>
+            <width>918</width>
+            <height>996</height>
            </rect>
           </property>
           <property name="autoFillBackground">
@@ -29361,8 +29361,8 @@ value as the Kp.</string>
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>980</width>
-            <height>1473</height>
+            <width>994</width>
+            <height>1466</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_9">
@@ -34777,6 +34777,13 @@ border-radius: 5;</string>
                    </size>
                   </property>
                  </spacer>
+                </item>
+                <item row="1" column="0" colspan="11">
+                 <widget class="QLabel" name="lblSwitchArmingWarning">
+                  <property name="text">
+                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;Warning:&lt;/span&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt; It is strongly recommended to use switch arming when HangTime is enabled.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+                  </property>
+                 </widget>
                 </item>
                </layout>
               </widget>

